### ISSUE TITLE
yt.load_sample() broken on macOS since commit 900667ac

### ISSUE THREAD

**Issue by rndsrc** [yt-project/yt#3227]
### Bug report

**Bug summary**

yt.load_sample() is broken since commit 900667ac "Merge branch 'main' into no_ambiguous_fields_automatic".

**Code for reproduction**

To reproduce, simply run the second cell of "yt/doc/source/quickstart/2)_Data_Inspection.ipynb".  Or,

```python
import yt
ds = yt.load_sample("IsolatedGalaxy")
```

The output is

```
yt : [INFO     ] 2021-04-27 13:37:38,888 'IsolatedGalaxy' is not available locally. Looking up online.
yt : [INFO     ] 2021-04-27 13:37:38,889 Downloading from https://yt-project.org/data/IsolatedGalaxy.tar.gz
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
<ipython-input-2-38f11556a912> in <module>
----> 1 ds = yt.load_sample("IsolatedGalaxy")

~/yt/yt/loaders.py in load_sample(fn, progressbar, timeout, **kwargs)
   1353     filename = urlsplit(specs["url"]).path.split("/")[-1]
   1354 
-> 1355     tmp_file = _download_sample_data_file(
   1356         filename, progressbar=progressbar, timeout=timeout
   1357     )

~/yt/yt/sample_data/api.py in _download_sample_data_file(filename, progressbar, timeout)
    195 
    196     poochie = _get_pooch_instance()
--> 197     poochie.fetch(filename, downloader=downloader)
    198     return Path.joinpath(poochie.path, filename)

/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pooch/core.py in fetch(self, fname, processor, downloader)
    518 
    519         # Create the local data directory if it doesn't already exist
--> 520         make_local_storage(str(self.abspath))
    521 
    522         url = self.get_url(fname)

/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pooch/utils.py in make_local_storage(path, env)
    280             # try to create the path at the same time. Use exist_ok to avoid
    281             # raising an error.
--> 282             os.makedirs(path, exist_ok=True)
    283         else:
    284             action = "write to"

/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/os.py in makedirs(name, mode, exist_ok)
    211     if head and tail and not path.exists(head):
    212         try:
--> 213             makedirs(head, exist_ok=exist_ok)
    214         except FileExistsError:
    215             # Defeats race condition when another thread created the path

/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/os.py in makedirs(name, mode, exist_ok)
    211     if head and tail and not path.exists(head):
    212         try:
--> 213             makedirs(head, exist_ok=exist_ok)
    214         except FileExistsError:
    215             # Defeats race condition when another thread created the path

/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/os.py in makedirs(name, mode, exist_ok)
    211     if head and tail and not path.exists(head):
    212         try:
--> 213             makedirs(head, exist_ok=exist_ok)
    214         except FileExistsError:
    215             # Defeats race condition when another thread created the path

/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/os.py in makedirs(name, mode, exist_ok)
    221             return
    222     try:
--> 223         mkdir(name, mode)
    224     except OSError:
    225         # Cannot rely on checking for EEXIST, since the operating system

OSError: [Errno 30] Read-only file system: '/does'
```

**Expected outcome**

`pooch` is not able to create the directory "/does" due to permission of the root file system and raised an OSError.  Most likely this is not a bug on `pooch` itself but incorrect setting of the download file location.

If I check out an earlier version: `git checkout 0c892879`, then the error is not there.

**Version Information**

  * Operating System: macOS 11.3
  * Python Version: 3.8
  * yt version: develop version, this occurs starting at commit 900667ac.  I checked ~ 5 different commits after 900667ac and they show the same error.
  * Other Libraries (if applicable): pooch v1.3.0, although this is likely a problem on the yt side.

This was discovered on macOS 11.3 with most of the yt dependencies installed by conda, and yt installed from source with development mode.

**Comment by neutrinoceros** [comment]
Hi @rndsrc . This looks a lot like the bug I fixed in #3195. Is you local copy of yt up to date ?
In the mean time, if you're still encountering this problem, a workaround is to set up you storage directory

The simplest way to do this is from the command line
```
yt config set test_data_dir ...
```
where you'll replace `...` with an existing directory of your choice. Hope that helps !

**Comment by neutrinoceros** [comment]
Sorry for taking so long to actually look at this. It was indeed still broken, but I got the fix now. Thank you very much for your early report, I appreciate it.
